pkg/reconciler/clusterducktype/clusterducktype.go

```diff
@@ -90,6 +100,21 @@ func (r *Reconciler) getCRDsWith(labelSelector string) ([]*apiextensionsv1.Custo
 	return list, nil
 }
 
+// resyncResourceMapper will make a call to the Kubernetes APIServer to request
+// the full list of resources on this cluster and then process the list to
+// create a lookup table between GroupVersions, Kinds and Resources.
+func (r *Reconciler) resyncResourceMapper(ctx context.Context) {
+	r.rmx.Lock()
+	defer r.rmx.Unlock()
```

Comment:
nit: why not add these ones just before line 115..

Reply:
it is to guard on the return for 113 also. so it would have to be in both places, or defer.

Reply:
hmm shouldn't you just guard what you are mutating here, i.e., the r.resourceMapper. Why do you need to guard the call to r.client.Discovery()... ?

Reply:
ah, I had not seen `RWMutex`. Nice I will use that.

Reply:
no, I don't think `RWMutex` is the right thing, I don't want to allow writing while reading. Let me try out the copying in the reconciler. 

let me work this a bit more in the usage PR.

Reply:
ok... I think when you held the write lock, readers get blocked... but readers don't block each other if there is no write lock held...
think about it in the follow up...